src/cmd/script_args/args.rs

```diff
@@ -0,0 +1,276 @@
+use anyhow::{format_err, Error, Result};
+use clap::Parser;
+use move_core_types::account_address::AccountAddress;
+use move_core_types::u256::U256;
+use serde::{Serialize, Serializer};
+use std::fmt;
+use std::str::FromStr;
+
+#[derive(Clone, Debug, Parser)]
+pub(super) struct ArgWithTypeVec {
+    /// Arguments combined with their type separated by spaces.
+    ///
+    /// Supported types [address, bool, hex, string, u8, u16, u32, u64, u128, u256, raw].
+    ///
+    /// Vectors may be specified using JSON array literal syntax (you may need to escape this with
+    /// quotes based on your shell interpreter).
+    ///
+    /// Example: `address:0x1 bool:true u8:0 u256:1234 "bool:[true, false]" 'address:[["0xace", "0xbee"], []]'`
+    #[clap(long, multiple_values(true))]
+    pub(super) args: Vec<ArgWithType>,
+}
+
+/// A parseable arg with a type separated by a colon.
+#[derive(Clone, Debug)]
+#[allow(dead_code)]
+pub(super) struct ArgWithType {
+    ty: FunctionArgType,
```

Comment:
Hm, unused actually - I just noticed this now in the morning. Let's practice some `backspace` here.

Reply:
Done